service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -43,4 +43,23 @@
      * A list of service URIs.
      */
     public abstract List<String> uris();
+
+    // hides implementation details
+    public static Builder builder() {
+        return ImmutableServiceConfiguration.builder();
+    }
+
+    // hides implementation details
+    public interface Builder {
```

Comment:
Uh, yeah i don't understand the reason for "hiding impl details" either. You still need to publish the Immutable* classes since that's the actual implementation, the only difference is that Immutable*.Builder is public....so what?